clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClientCallbackHandler.java

```diff
@@ -97,7 +97,7 @@ public void close() {
     private void handleCallback(OAuthBearerTokenCallback callback) throws IOException {
         if (callback.token() != null)
             throw new IllegalArgumentException("Callback had a token already");
-        Subject subject = Subject.getSubject(AccessController.getContext());
+        Subject subject = SecurityManagerCompatibility.get().current();
         Set<OAuthBearerToken> privateCredentials = subject != null
```

Comment:
Few tests are failing and seems related due to `privateCredentials` being empty when active strategy is `LegacyStratagy`

Reply:
Thanks, this was due to doAs throwing a checked exception that unintentionally got wrapped in a RuntimeException. This should be fixed now.